### DISCUSSION TITLE
Need help with converting old codes to newer version

### DISCUSSION THREAD

**Discussion by wwang86** [milaboratory/mixcr#1507] [Q&A] [answered]
Hi, I am really new to MiXCR and have no experience with bio-informatic analysis. I have a code from our previous lab member which was used for ver 3.0.13. I want to see if anyone can help me update the code to be used on ver 4.6.0. 

`mixcr analyze amplicon -s mmu --starting-material rna --5-end v-primers --3-end j-primers --adapters adapters-present --assemble "-OassemblingFeatures=[VDJRegion]" --export "-p fullImputed" Input_R1.fastq.gz Input_R2.fastq.gz output`

**Comment by mizraelson** [comment, marked as answer]
Hi,
So the updated code will look like this:

```
mixcr analyze generic-amplicon \
    --species mmu \
    --rna \
    --floating-left-alignment-boundary \
    --floating-right-alignment-boundary J \
    --assemble-clonotypes-by "VDJRegion" \
    --impute-germline-on-export \
      input_R1.fastq.gz \
      input_R2.fastq.gz \
      result
```

**Reply by wwang86** [reply]
Hi, thank you for the reply! However, I am still having trouble getting exactly the same result as the old version. Specifically, if I do not include `assemble-clonotypes-by "VDJRegion`, I can at least get exactly the same CDR3 sequences as I got using the old version, but it won't give me the rest of the antibody sequences. If I include the command, the number of assmebled clonotypes decreases dramatically and gives me completely different result. Could you further help me with this? Thank you so much!

**Reply by mizraelson** [reply]
The algorithms under the hood changed dramatically since v 3.03 thus the results may differ. In general now they are way more reliable. If you can show an example of the difference I can help you back track it to the original reads to show why the current version gave a certain result.
